src/Microsoft.AspNetCore.Sockets.Client/Connection.cs

```diff
@@ -194,20 +193,21 @@ public Task<bool> SendAsync(byte[] data, MessageType type)
 
             if (_connectionState != ConnectionState.Connected)
             {
-                return false;
+                throw new InvalidOperationException(
+                    $"Cannot send messages when the connection is not in the Connected state.");
             }
 
-            var message = new Message(data, type);
+            var sendTcs = new TaskCompletionSource<object>();
+            var message = new SendMessage(data, type, sendTcs);
 
             while (await Output.WaitToWriteAsync(cancellationToken))
             {
                 if (Output.TryWrite(message))
                 {
-                    return true;
+                    await sendTcs.Task;
```

Comment:
The purpose of this change is to be able to tell the user whether we were able to to send their message. Therefore we need to wait until the message is sent before we can return. I guess they don't have to await the call if they are not interested in the result they would just not await the call.